lib/usd/translators/shading/usdUVTextureReader.cpp

```diff
@@ -215,42 +220,197 @@ bool PxrMayaUsdUVTexture_Reader::Read(UsdMayaPrimReaderContext* context)
+            char* fileDataBuf = (char*)malloc(sizeof(char) * fileSize);
+            memcpy(fileDataBuf, fileData.get(), fileSize);
+            uint64_t spookyHash = ArchHash64(fileDataBuf, fileSize);
+            free(fileDataBuf);
+            std::unordered_map<std::string, uint64_t>::iterator itExistingHash
+                = UsdMayaReadUtil::mapFileHashes.find(unresolvedFilePath);
+            if (itExistingHash
+                == UsdMayaReadUtil::mapFileHashes.end()) { // NOTE: (yliangsiew) Means that the
+                                                           // texture hasn't been extracted before.
+                UsdMayaReadUtil::mapFileHashes.insert(
+                    { unresolvedFilePath,
+                      spookyHash }); // NOTE: (yliangsiew) This _should_ be the common case.
+            } else {
+                uint64_t existingHash = itExistingHash->second;
+                if (spookyHash == existingHash) {
+                    TF_WARN(
+                        "A duplicate texture: %s was found, skipping extraction of it and re-using "
+                        "the existing one.",
+                        unresolvedFilePath.c_str());
+                    unresolvedFilePath = itExistingHash->first;
+                } else {
+                    // NOTE: (yliangsiew) Means that a duplicate texture with the same name but with
+                    // different contents was found. Instead of failing, continue extraction with a
+                    // different filename instead and point to that one.
+                    needsUniqueFilename = true;
                 }
             }
-            UsdMayaReadUtil::SetMayaAttr(mayaAttr, val);
+
+            // NOTE: (yliangsiew) Write the file to disk now.
+            std::string filename(unresolvedFilePath);
+            UsdMayaUtilFileSystem::pathStripPath(filename);
+            std::string extractedFilePath(jobArgs.importUSDZTexturesFilePath);
+            bool        bStat = UsdMayaUtilFileSystem::pathAppendPath(extractedFilePath, filename);
+            TF_VERIFY(bStat);
+
+            if (needsUniqueFilename) {
+                int         counter = 0;
+                std::string checkPath(extractedFilePath);
+                while (ghc::filesystem::is_regular_file(checkPath)) {
+                    checkPath.assign(extractedFilePath);
+                    std::string filenameNoExt(checkPath);
+                    std::string ext = UsdMayaUtilFileSystem::pathFindExtension(checkPath);
+                    UsdMayaUtilFileSystem::pathRemoveExtension(checkPath);
+                    char newFileName[FILENAME_MAX] = { 0 };
+                    int  numChars = snprintf(
+                        newFileName,
+                        FILENAME_MAX,
+                        "%s_%d%s",
+                        checkPath.c_str(),
+                        counter,
+                        ext.c_str());
+                    TF_VERIFY(numChars > 0);
```

Comment:
Please stay in C++:
extractedFilePath = checkPath + boost::lexical_cast<std::string>(counter) + ext;
Or use USD APIs:
extractedFilePath = TfStringPrintf("%s_%d%s", checkPath.c_str(), counter, ext.c_str());